### PR TITLE
fix: Move min-height styles from Box to FieldMessage

### PR DESCRIPTION
**Pull request by markdalgleish**
## Breaking changes

- `Box` no longer has a `minHeight` prop.

## Development notes

I've introduced a `utils` collection that's passed into each themed block of CSS which contains `desktopStyles`, `responsiveStyles` and `rows` functions.

I've also refactored Icon styles to use theme utils where possible.

### CODE REVIEW

File: lib/atoms/makeSizes.ts [hunk 1]
```diff
@@ -1,47 +1,7 @@
...
-  return merge({}, ...rules);
-};
 
 const width = {
   '.width_full': { width: '100%' },
 };
 
-export default (tokens: Tokens) =>
-  merge({}, makeSizeRules(tokens, 'minHeight'), width);
+export default () => merge({}, width);
```
Comment (mattsjones) [inline lib/atoms/makeSizes.ts]: Happy to leave the merge? lol

Comment (michaeltaranto) [inline lib/atoms/makeSizes.ts]: The merge is probably overkill here now.

File: lib/components/FieldMessage/FieldMessage.treat.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    justifyContent: 'flex-end',
+  },
+  'root',
+);
+
+export const grow = style(
+  {
+    flexGrow: 1,
+  },
+  'grow',
```
Comment (mattsjones) [inline lib/components/FieldMessage/FieldMessage.treat.ts]: If you used `css` instead we wouldn't need the debug name here. Personally, think that would be cleaner.

File: lib/components/FieldMessage/FieldMessage.tsx [hunk 3]
```diff
@@ -25,7 +26,7 @@ const renderIcon = (tone: FieldTone = 'neutral') => {
   };
 
   return (
-    <Box paddingRight="xsmall" className={styles.fixedSize}>
+    <Box paddingRight="xsmall" className={useClassNames(styles.fixedSize)}>
```
Comment (mattsjones) [inline lib/components/FieldMessage/FieldMessage.tsx]: I think `useStyles` would be cleaner here.

File: lib/components/FieldMessage/FieldMessage.treat.ts (hunk 2 above)
Comment (markdalgleish) [reply lib/components/FieldMessage/FieldMessage.treat.ts]: As discussed, gonna leave this it as-is until we optimise the output of themed `css` blocks.

File: lib/components/FieldMessage/FieldMessage.tsx (hunk 3 above)
Comment (markdalgleish) [reply lib/components/FieldMessage/FieldMessage.tsx]: Incorrect.

**Pull request merged**